src/libraries/System.IO.Pipes/src/System/IO/Pipes/NamedPipeClientStream.cs

```diff
@@ -194,7 +194,12 @@ public Task ConnectAsync(int timeout, CancellationToken cancellationToken)
             }
 
             int startTime = Environment.TickCount; // We need to measure time here, not in the lambda
-            return Task.Run(() => ConnectInternal(timeout, cancellationToken, startTime), cancellationToken);
+
+            return Task.Factory.StartNew(static state =>
```

Comment:
Why Task.Factory.StartNew -- so you could pass DenyChildAttach?

Reply:
I believe it's just because `Task.Factory.StartNew` accepts a `state` object, whereas `Task.Run` doesn't.

Reply:
> because Task.Factory.StartNew accepts a state object

Right